nodes/BraveSearch/utils/errorHandler.ts

```diff
@@ -0,0 +1,211 @@
+import { NodeApiError, type INode } from 'n8n-workflow';
+
+export interface BraveSearchApiError {
+    id: string;
+    status: number;
+    code: string;
+    detail: string;
+    meta?: {
+        errors?: Array<{
+            type: string;
+            loc: string[];
+            msg: string;
+            input?: string;
+            ctx?: {
+                expected?: string;
+            };
+        }>;
+    };
+}
+
+export interface BraveSearchErrorResponse {
+    type: string;
+    error: BraveSearchApiError;
+    time: number;
+}
+
+export class BraveSearchErrorHandler {
+    /**
+     * Creates a user-friendly error message from Brave Search API error response
+     */
```

Comment:
How has this code been tested? Do you have a reliable method for invoking error responses?